modules/ml/src/tree.cpp

```diff
@@ -48,6 +48,50 @@ namespace ml {
 
 using std::vector;
 
+struct ParallelBestSplit
+{
+    ParallelBestSplit (const vector<int>& sidx, DTreesImpl * dtree)
+            : tree(dtree), _sidx(sidx)
+    {
+        best_split = new DTreesImpl::WSplit();
+        best_split->quality = 0.;
+    }
+
+    ~ParallelBestSplit()
+    {
+        delete best_split;
+    }
+    void operator ()(const cv::BlockedRange& range)
+    {
+        DTreesImpl::WSplit split;
+        const std::vector<int> activeVars = tree->getActiveVars();
+
+        AutoBuffer<int> buf(tree->w->maxSubsetSize*2);
+        int *subset = buf.data(), *best_subset = subset + tree->w->maxSubsetSize;
+        for (int vi_ = range.begin(); vi_ < range.end(); vi_++) {
+            int vi = activeVars[vi_];
+            if (tree->varType[vi] == VAR_CATEGORICAL) {
+                if (tree->_isClassifier)
+                    split = tree->findSplitCatClass(vi, _sidx, 0, subset);
+                else
+                    split = tree->findSplitCatReg(vi, _sidx, 0, subset);
+            } else {
+                if (tree->_isClassifier)
+                    split = tree->findSplitOrdClass(vi, _sidx, 0);
+                else
+                    split = tree->findSplitOrdReg(vi, _sidx, 0);
+            }
+            if (split.quality > best_split->quality) {
+                memcpy(best_split, &split, sizeof(split));
+                std::swap(subset, best_subset);
+            }
+        }
+    }
+    DTreesImpl * tree;
+    vector<int> _sidx;
+    DTreesImpl::WSplit * best_split;
```

Comment:
`WSplit` is just structure with several fields. It's more efficient just add field of type `DTreesImpl::WSplit`, but not pointer. No need to do new/delete each time.

Reply:
done!